src/SessionTracker.php

```diff
@@ -0,0 +1,326 @@
+     * The current payload version.
+     */
+    protected static $SESSION_PAYLOAD_VERSION = '1.0';
+
+    /**
+     * The current client configuration.
+     *
+     * @var Configuration
+     */
+    protected $config;
+
+    /**
+     * An array of session counts.
+     *
+     * @var array
+     */
+    protected $sessionCounts = [];
+
+    /**
+     * A locking function for synchronisation.
+     *
+     * @var function
+     */
+    protected $lockFunction = null;
+
+    /**
+     * An unlocking function for synchronisation.
+     *
+     * @var function
+     */
+    protected $unlockFunction = null;
+
+    /**
+     * A function to use when retrying a failed delivery.
+     *
+     * @var function
+     */
+    protected $retryFunction = null;
+
+    /**
+     * A function to store/get data.
+     *
+     * @var function
+     */
+    protected $storageFunction = null;
+
+    /**
+     * A function to store/get sessions.
+     *
+     * @var function
+     */
+    protected $sessionFunction = null;
+
+    /**
+     * The last time the sessions were delivered.
+     *
+     * @var int
+     */
+    protected $lastSent;
+
+    /**
+     * The current session.
+     *
+     * @var array
+     */
+    protected $currentSession;
+
+    /**
+     * Create a session tracker instance.
+     *
+     * @param Configuration $config the initial client configuration
+     *
+     * @return void
+     */
+    public function __construct(Configuration $config)
+    {
+        $this->config = $config;
+        $this->lastSent = 0;
+    }
+
+    public function setConfig(Configuration $config)
+    {
+        $this->config = $config;
+    }
+
+    public function createSession()
+    {
+        if (!$this->config->shouldTrackSessions()) {
+            return;
+        }
+        $currentTime = strftime('%Y-%m-%dT%H:%M:00');
+        $session = [
+            'id' => uniqid('', true),
+            'startedAt' => $currentTime,
+            'events' => [
+                'handled' => 0,
+                'unhandled' => 0,
+            ],
+        ];
+        $this->setCurrentSession($session);
+        $this->incrementSessions($currentTime);
+    }
+
+    protected function setCurrentSession(array $session)
+    {
+        if (!is_null($this->sessionFunction)) {
+            call_user_func($this->sessionFunction, $session);
+        } else {
+            $this->currentSession = $session;
+        }
+    }
+
+    public function getCurrentSession()
+    {
+        if (!is_null($this->sessionFunction)) {
+            return call_user_func($this->sessionFunction);
+        } else {
+            return $this->currentSession;
+        }
+    }
+
+    public function sendSessions()
+    {
+        $locked = false;
+        if (!is_null($this->lockFunction) && !is_null($this->unlockFunction)) {
+            call_user_func($this->lockFunction);
+            $locked = true;
+        }
+
+        try {
+            $this->deliverSessions();
+        } finally {
```

Comment:
Another use of php 5.5+ features here.

Reply:
As I understand our minimum supported version is 5.5.9, so this shouldn't be an issue?

Reply:
Oh, yeh, the minimum is 5.5.0 in this package. Looking at the history It was bumped from 5.2.0 to 5.5.9 in the v3.0.0 release, then moved down to 5.5.0 in v3.0.1+.

Reply:
How do you dismiss reviews in GitHub? :P